src/notification.c

```diff
@@ -429,41 +473,25 @@ void notification_init(notification *n)
                 g_free(n->msg);
                 n->msg = buffer;
         }
+}
 
-        n->dup_count = 0;
-
-        /* urgency > CRIT -> array out of range */
-        n->urgency = n->urgency > CRIT ? CRIT : n->urgency;
-
-        if (!n->colors[ColFG]) {
-                n->colors[ColFG] = xctx.colors[ColFG][n->urgency];
-        }
-
-        if (!n->colors[ColBG]) {
-                n->colors[ColBG] = xctx.colors[ColBG][n->urgency];
-        }
-
-        if (!n->colors[ColFrame]) {
-                n->colors[ColFrame] = xctx.colors[ColFrame][n->urgency];
-        }
-
-        n->timeout =
-            n->timeout < 0 ? settings.timeouts[n->urgency] : n->timeout;
-        n->start = 0;
-
-        n->timestamp = g_get_monotonic_time();
-
-        n->redisplayed = false;
-
-        n->first_render = true;
+static void notification_extract_urls(notification *n)
+{
+        // DO markup urls processing here until we split this out correctly
+        n->urls = notification_extract_markup_urls(&(n->body)); //WHAT THE FUCK!?
```

Comment:
I'm not sure what the `WTF` is here for? - Sure not the best code but not really weird?

Reply:
Thank you for mentioning this again.

For me, this function call is against every logic. Why do we put a double pointer to that (and not the notification and turn it into a method like every other method).

I also don't see, why we do have to make changes to the body field there. Markup should be extracted somewhere else.

Reply:
1. There is a discrepancy between pango and notification spec. Notification spec requires `img` and `a` tags, while pango cannot render these (obviously). As `a` tags happen quite regularly, these have to get replaced. Guess, what this method did, too!?

2. Let's have these two notifications:
    - `notify-send asdf '<a href="https://github.com/">testname</a>'`
        - ` n->urls = 0x5555558675f0 "[#1] https://github.com/"`
    - `notify-send https://summary-url.de https://internal.com`
        - `1: n->urls = 0x555555803980 "https://summary-url.de\nhttps://internal.com"`
    - Mhmm, same field, different format, 👎